core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -128,26 +128,30 @@
     public static void verifySourceAggregations(
             List<Expression> groupByExpressions,
             Scope sourceScope,
-            Expression expression,
+            List<Expression> expressions,
             Session session,
             Metadata metadata,
             Analysis analysis)
     {
         AggregationAnalyzer analyzer = new AggregationAnalyzer(groupByExpressions, sourceScope, Optional.empty(), session, metadata, analysis);
-        analyzer.analyze(expression);
+        for (Expression expression : expressions) {
+            analyzer.analyze(expression);
+        }
     }
 
     public static void verifyOrderByAggregations(
             List<Expression> groupByExpressions,
             Scope sourceScope,
             Scope orderByScope,
-            Expression expression,
+            List<Expression> expressions,
```

Comment:
Can we return from this method early if list of `expressions` is empty? Without creation of `AggregationAnalyzer`. Then you don't need to have this `if` in `analyzeAggregations`.

Reply:
`if` in analyzeExpression ensure that we use `orderBy` scope only for verifying orderBy aggregations.